java-lib/src/main/java/com/wavefront/predicates/EvalExpressionVisitorImpl.java

```diff
@@ -0,0 +1,471 @@
+    return (StringExpression) entity ->
+        isTrue(condition.getValue(entity)) ?
+            thenExpression.getString(entity) :
+            elseExpression.getString(entity);
+  }
+
+  private EvalExpression eval(EvalExpressionParser.EvalExpressionContext ctx) {
+    return (EvalExpression) visitEvalExpression(ctx);
+  }
+
+  private StringExpression stringExpression(EvalExpressionParser.StringExpressionContext ctx) {
+    return (StringExpression) visitStringExpression(ctx);
+  }
+
+  private EvalExpression iff(EvalExpression cond, EvalExpression thenExp, EvalExpression elseExp) {
+    return x -> isTrue(cond.getValue(x)) ? thenExp.getValue(x) : elseExp.getValue(x);
+  }
+
+  private static double getNumber(EvalExpressionParser.NumberContext numberContext) {
+    if (numberContext == null) {
+      throw new IllegalArgumentException("Cannot parse number (null)");
+    }
+    if (numberContext.Number() == null) {
+      throw new IllegalArgumentException("Cannot parse number from: \"" +
+          numberContext.getText() + "\" " + numberContext.getText());
+    }
+    String text = numberContext.Number().getText();
+    double toReturn = text.startsWith("0x") ? Long.decode(text) : Double.parseDouble(text);
+    if (numberContext.MinusSign() != null) {
+      toReturn = -toReturn;
+    }
+    if (numberContext.siSuffix() != null) {
+      String suffix = numberContext.siSuffix().getText();
+      switch (suffix) {
+        case "Y":
+          toReturn *= 1E24;
+          break;
+        case "Z":
+          toReturn *= 1E21;
+          break;
+        case "E":
+          toReturn *= 1E18;
+          break;
+        case "P":
+          toReturn *= 1E15;
+          break;
+        case "T":
+          toReturn *= 1E12;
+          break;
+        case "G":
+          toReturn *= 1E9;
+          break;
+        case "M":
+          toReturn *= 1E6;
+          break;
+        case "k":
+          toReturn *= 1E3;
+          break;
+        case "h":
+          toReturn *= 1E2;
+          break;
+        case "da":
+          toReturn *= 10;
+          break;
+        case "d":
+          toReturn *= 1E-1;
+          break;
+        case "c":
+          toReturn *= 1E-2;
+          break;
+        case "m":
+          toReturn *= 1E-3;
+          break;
+        case "µ":
+          toReturn *= 1E-6;
+          break;
+        case "n":
+          toReturn *= 1E-9;
+          break;
+        case "p":
+          toReturn *= 1E-12;
+          break;
+        case "f":
+          toReturn *= 1E-15;
+          break;
+        case "a":
+          toReturn *= 1E-18;
+          break;
+        case "z":
+          toReturn *= 1E-21;
+          break;
+        case "y":
+          toReturn *= 1E-24;
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown SI Suffix: " + suffix);
+      }
+    }
+    return toReturn;
+  }
+
+  private static int murmurhash3_x86_32(CharSequence data) {
```

Comment:
Can you use the Guava hash function for this (not as efficient but probably more maintainable?)

Reply:
Done